tests/phpunit/tests/test-ajax-on-front.php

```diff
@@ -28,26 +30,81 @@ public function _ajax_test_locale() {
 		wp_die( wp_json_encode( __( 'Dashboard' ) ) ); // phpcs:ignore WordPress.WP.I18n.MissingArgDomain
 	}
 
-	public function test_locale_for_logged_in_user() {
-		wp_set_current_user( 1 );
+	/**
+	 * @dataProvider ajax_on_front_provider
+	 *
+	 * @param string $language    Language code of the ajax request.
+	 * @param string $translation "Dashboard" translated in the current language.
+	 * @param int    $user_id     1 of logged in user, 0 for anonymous user.
+	 * @param string $source      Source to set the language: 'param' or 'cookie'.
+	 */
+	public function test_ajax_on_front( $language, $translation, $user_id, $source ) {
 		update_user_meta( 1, 'locale', 'en_US' );
 
-		$links_model = self::$model->get_links_model();
-		$frontend = new PLL_Frontend( $links_model );
-		$frontend->curlang = self::$model->get_language( 'fr' );
-		new PLL_Frontend_Filters( $frontend );
+		wp_set_current_user( $user_id );
 
-		add_action( 'wp_ajax_test_locale', array( $this, '_ajax_test_locale' ) );
+		$_REQUEST = array( 'action' => 'test' );
 
-		$_REQUEST['action'] = 'test_locale';
+		if ( 'cookie' === $source ) {
+			$_COOKIE['pll_language'] = $language;
+		} else {
+			$_REQUEST['lang'] = $language;
+		}
+
+		add_action( 'wp_ajax_test', array( $this, '_ajax_test_locale' ) );
+
+		$links_model = self::$model->get_links_model();
+		$frontend = new PLL_Frontend( $links_model );
+		$frontend->init();
 
 		try {
-			$this->_handleAjax( 'test_locale' );
+			$this->_handleAjax( 'test' );
 		} catch ( WPAjaxDieStopException $e ) {
```

Comment:
```suggestion
} catch ( WPAjaxDieContinueException $e ) {
			unset( $e );
		}
		$response = json_decode( $this->_last_response, true );
```
Follows my previous comment.

Reply:
Done in a1aae1c67a9baec6fbbc1704a0843e3aa691abbf